### ISSUE TITLE
Wildcard Starting Value in Search String Causes API Error

### ISSUE THREAD

**Issue by dthagard** [Graylog2/graylog2-web-interface#1140]
When performing a search, if the query starts with an asterisk (*) followed by any number of non-white space characters, an API error is thrown. The error occurs if any search entry starts with a wildcard.

```
(You caused a org.graylog2.restclient.lib.APIException. API call failed GET http://@10.10.9.233:12900/search/universal/relative/histogram?range=300&interval=minute&query=*a&range_type=relative&filter=streams:54edcfdde4b0109271e2913d returned 400 Bad Request body: {"query":"*a","begin_column":0,"begin_line":0,"end_column":0,"end_line":0})

Stacktrace
org.graylog2.restclient.lib.ApiClientImpl$ApiRequestBuilder#execute (ApiClientImpl.java:481)
org.graylog2.restclient.models.UniversalSearch#dateHistogram (UniversalSearch.java:185)
controllers.SearchController#renderSearch (SearchController.java:130)
controllers.StreamSearchController#index (StreamSearchController.java:57)
Routes$$anonfun$routes$1$$anonfun$applyOrElse$14$$anonfun$apply$404#apply (routes_routing.scala:1437)
Routes$$anonfun$routes$1$$anonfun$applyOrElse$14$$anonfun$apply$404#apply (routes_routing.scala:1437)
play.core.Router$HandlerInvokerFactory$$anon$4#resultCall (Router.scala:264)
play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1#invocation (Router.scala:255)
play.core.j.JavaAction$$anon$1#call (JavaAction.scala:55)
play.GlobalSettings$1#call (GlobalSettings.java:67)
play.mvc.Security$AuthenticatedAction#call (Security.java:44)
play.core.j.JavaAction$$anonfun$11#apply (JavaAction.scala:82)
play.core.j.JavaAction$$anonfun$11#apply (JavaAction.scala:82)
scala.concurrent.impl.Future$PromiseCompletingRunnable#liftedTree1$1 (Future.scala:24)
scala.concurrent.impl.Future$PromiseCompletingRunnable#run (Future.scala:24)
play.core.j.HttpExecutionContext$$anon$2#run (HttpExecutionContext.scala:40)
play.api.libs.iteratee.Execution$trampoline$#execute (Execution.scala:46)
play.core.j.HttpExecutionContext#execute (HttpExecutionContext.scala:32)
scala.concurrent.impl.Future$#apply (Future.scala:31)
scala.concurrent.Future$#apply (Future.scala:485)
play.core.j.JavaAction$class#apply (JavaAction.scala:82)
play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1#apply (Router.scala:252)
play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5#apply (Action.scala:130)
play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5#apply (Action.scala:130)
play.utils.Threads$#withContextClassLoader (Threads.scala:21)
play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4#apply (Action.scala:129)
play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4#apply (Action.scala:128)
scala.Option#map (Option.scala:145)
play.api.mvc.Action$$anonfun$apply$1#apply (Action.scala:128)
play.api.mvc.Action$$anonfun$apply$1#apply (Action.scala:121)
play.api.libs.iteratee.Iteratee$$anonfun$mapM$1#apply (Iteratee.scala:483)
play.api.libs.iteratee.Iteratee$$anonfun$mapM$1#apply (Iteratee.scala:483)
play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1#apply (Iteratee.scala:519)
play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1#apply (Iteratee.scala:519)
play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14#apply (Iteratee.scala:496)
play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14#apply (Iteratee.scala:496)
scala.concurrent.impl.Future$PromiseCompletingRunnable#liftedTree1$1 (Future.scala:24)
scala.concurrent.impl.Future$PromiseCompletingRunnable#run (Future.scala:24)
akka.dispatch.TaskInvocation#run (AbstractDispatcher.scala:41)
akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask#exec (AbstractDispatcher.scala:393)
scala.concurrent.forkjoin.ForkJoinTask#doExec (ForkJoinTask.java:260)
scala.concurrent.forkjoin.ForkJoinPool$WorkQueue#runTask (ForkJoinPool.java:1339)
scala.concurrent.forkjoin.ForkJoinPool#runWorker (ForkJoinPool.java:1979)
scala.concurrent.forkjoin.ForkJoinWorkerThread#run (ForkJoinWorkerThread.java:107)
Request information
Method
GET
Query
relative=300
to=
q=*a
fields=source,message
from=
rangetype=relative
width=1903
X-Forwarded-For
10.1.12.60
Connection
close
Accept
text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
X-Real-IP
10.1.12.60
Accept-Language
en-US,en;q=0.8
Accept-Encoding
gzip, deflate, sdch
User-Agent
Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.115 Safari/537.36
Host
logs.mydomain.com:443
X_FORWARDED_PROTO
https
```

**Comment by edmundoa** [comment]
Hi,

Which Graylog version are you using? This looks similar to https://github.com/Graylog2/graylog2-server/issues/947, which should be fixed in 1.0.

By the way, did you enable leading wildcards in your Graylog server configuration file? You need to do so in order to use leading wildcards: http://docs.graylog.org/en/1.0/pages/queries.html#syntax

**Comment by edmundoa** [comment]
We haven't heard from you in a while, so I will close the issue. Please reopen if it's still valid.
